src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,315 @@
+        }
+        return [makePxValue(1.25), theme.spacings.xxlarge];
+      default:
+        if (!children) {
+          return [theme.spacings.small];
+        }
+        return [theme.spacings.small, theme.spacings.xxlarge];
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return 'small';
+        }
+        return 'xsmall';
+      case 'small':
+        if (!children) {
+          return 'medium';
+        }
+        return 'small';
+      case 'medium':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      case 'large':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      default:
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        return iconAlign === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'small':
+        return iconAlign === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'medium':
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+      case 'large':
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+      default:
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  borderRadius({ theme }) {
+    return theme.spacings.xxsmall;
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${styles.borderRadius};
+  border: ${styles.border};
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  testID,
+}) => {
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw Error("Only string is allowed as Button's children");
+  }
+
+  const theme = useContext(ThemeContext);
```

Comment:
define all the variables before any conditions or logic for the component. Move this before the `if` check